942.dI-string-match.js

```diff
@@ -0,0 +1,30 @@
+/**
+ *https://leetcode.com/problems/di-string-match/description/
+ */
+/**
+ * @param {string} s
+ * @return {number[]}
+ */
+var diStringMatch = function (s) {
+  const arr = [];
+
+  for (let i = 0; i < s.length + 1; i++) {
+    arr.push(i);
+  }
```

Comment:
```suggestion
  for (let i = 0; i <= s.length; i++) {
    arr.push(i);
  }
```

I think you can define startPoint and endPoint like below.
`let startPoint = 0`
`let endPoint = s.length`
Then code and logic can be more simple